src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTabList.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+/**
+ * This software is provided under the terms of the Minecraft Forge Public
+ * License v1.0.
+ */
+package net.minecraftforge.client.player.inventory.tabs;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.gui.inventory.GuiInventory;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.client.config.GuiUtils;
+
+/**
+ * Enclosing class to keep GuiContainer clean. This handles all the survival tab drawing, and the addition of two buttons.
+ */
+public class GuiTabList
+{
+	private int page;
+	private int maxPages;
+	private GuiContainer container;
+	private List<GuiTab> tabs;
+	private static final ResourceLocation TEXTURE_TABS = new ResourceLocation("textures/gui/container/creative_inventory/tabs.png");
+
+	public GuiTabList(GuiContainer container)
+	{
+		this.container = container;
+		tabs = new ArrayList<GuiTab>(GuiTab.getTabsForGui(container.getClass())); //get a copy of the stored list so it can be modified without consequence
+		maxPages = ((getTabs().size() - 12) / 12) + 1;
+
+		for (GuiTab tab : getTabs())//set the first page
+			if (container.getClass().equals(tab.getTargetGui()))
```

Comment:
Try only calling container.getClass() once ?
I think since you create one list for each gui, better be efficient at creation time.